### PR TITLE
#76533 CI pipeline

### PR DESCRIPTION
**Pull request by A-Herold**
## PR Type

[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no API changes)
[ ] Build-related changes
[ x ] CI-related changes
[ ] Documentation content changes
[ ] Application / infrastructure changes
[ ] Other: 

## What Is the Current Behavior?

Issue Number: Closes #

## What Is the New Behavior?

CI pipeline in Azure DevOps will be triggered on every git push.

## Does this PR Introduce a Breaking Change?

[ x ] Yes
[ ] No

The maven repo, providing the IOM artifacts, was changed. To be able to get access to this repo, ~/.m2/settings.xml has to be extended by an additional server:

`
    <server>
      <id>order-iom-blueprint</id>
      <username>[same username as of order-iom-releases]</username>
      <password>[same password as of order-iom-releases]</password>
    </server>
`

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# see https://docs.microsoft.com/en-us/azure/devops/pipelines/process/resources?view=azure-devops&tabs=schema#define-a-repositories-resource
+resources:
+  repositories:
+    - repository: IOMDevopsRepo
+      type: git
+      name: order-iom-partner-devops
+      ref: task/76533-ci-pipeline
```
Comment (jlistemann-ish) [inline azure-pipelines.yml:7]: this fallback is not for prod, right ? and will be deleted after ...

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# see https://docs.microsoft.com/en-us/azure/devops/pipelines/process/resources?view=azure-devops&tabs=schema#define-a-repositories-resource
```
Comment (jlistemann-ish) [inline azure-pipelines.yml:1]: Good idea to put the links to the official doc here.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  repositories:
+    - repository: IOMDevopsRepo
+      type: git
+      name: order-iom-partner-devops
+      ref: task/76533-ci-pipeline
+
+trigger:
+  branches:
+    include:
+      - '*'
```
Comment (jlistemann-ish) [inline azure-pipelines.yml:12]: as I understand, triggered by any change in any branch

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+# see https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#extend-from-a-template
+# see https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#use-other-repositories
+extends:
+  template: template.yml@IOMDevopsRepo
```
Comment (jlistemann-ish) [inline azure-pipelines.yml:17]: Good idea to extend from a compony owned template.
These parameters are defined by us and are not reserved words, right ?

File: azure-pipelines.yml (hunk 1 above)
Comment (A-Herold) [reply azure-pipelines.yml:7]: All projects have use the main branch of the pipeline template. But currently there is no pipeline template in main. This has to be changed later, when template is merged to its final position.

File: azure-pipelines.yml (hunk 3 above)
Comment (A-Herold) [reply azure-pipelines.yml:12]: Yes. But this part of the pipeline is fully in the hands of partners (you are the partner of the Blueprint project). The partner can simply change this setting.

File: azure-pipelines.yml (hunk 4 above)
Comment (A-Herold) [reply azure-pipelines.yml:17]: Right

**Review by Philipp-Borchert-ISH — APPROVED** [review]

**Pull request merged**